src/coff_import_file.rs

```diff
@@ -3,7 +3,898 @@
+          // .idata$3
+          size_of::<ImageImportDescriptor>()).try_into().unwrap()),
+      number_of_symbols: u32!(NUMBER_OF_SYMBOLS.try_into().unwrap()),
+      size_of_optional_header: u16!(0),
+      characteristics: u16!(if self.is_64_bit() { 0 } else {IMAGE_FILE_32BIT_MACHINE }),
+  };
+        buffer.write_all(unsafe { any_as_u8_slice(&header) })?;
+
+        // Section Header Table
+        let section_table: [_; NUMBER_OF_SECTIONS] = [ImageSectionHeader {
+            name: *b".idata$3",
+            virtual_size: u32!(0),
+            virtual_address: u32!(0),
+            size_of_raw_data: u32!(size_of::<ImageImportDescriptor>().try_into().unwrap()),
+            pointer_to_raw_data: u32!((size_of::<ImageFileHeader>()
+                + NUMBER_OF_SECTIONS * size_of::<ImageSectionHeader>())
+            .try_into()
+            .unwrap()),
+            pointer_to_relocations: u32!(0),
+            pointer_to_linenumbers: u32!(0),
+            number_of_relocations: u16!(0),
+            number_of_linenumbers: u16!(0),
+            characteristics: u32!(
+                IMAGE_SCN_ALIGN_4BYTES
+                    | IMAGE_SCN_CNT_INITIALIZED_DATA
+                    | IMAGE_SCN_MEM_READ
+                    | IMAGE_SCN_MEM_WRITE
+            ),
+        }];
+        buffer.write_all(unsafe { any_as_u8_slice(&section_table) })?;
+
+        // .idata$3
+        let import_descriptor = ImageImportDescriptor {
+            original_first_thunk: u32!(0),
+            time_date_stamp: u32!(0),
+            forwarder_chain: u32!(0),
+            name: u32!(0),
+            first_thunk: u32!(0),
+        };
+        buffer.write_all(unsafe { any_as_u8_slice(&import_descriptor) })?;
+
+        // Symbol Table
+        let mut symbol_table: [_; NUMBER_OF_SYMBOLS] = [ImageSymbol {
+            name: [0; 8],
+            value: u32!(0),
+            section_number: u16!(1),
+            typ: u16!(0),
+            storage_class: IMAGE_SYM_CLASS_EXTERNAL,
+            number_of_aux_symbols: 0,
+        }];
+        set_name_to_string_table_entry(&mut symbol_table[0], size_of::<u32>());
+        buffer.write_all(unsafe { any_as_u8_slice(&symbol_table) })?;
+
+        // String Table
+        write_string_table(&mut buffer, &[NULL_IMPORT_DESCRIPTOR_SYMBOL_NAME])?;
+
+        Ok(NewArchiveMember::new(
+            buffer.into_boxed_slice(),
+            &DEFAULT_OBJECT_READER,
+            self.import_name.to_string(),
+        ))
+    }
+
+    /// Create a NULL Thunk Entry.  This is a small object file which contains a
+    /// NULL Import Address Table entry and a NULL Import Lookup Table Entry.  It
+    /// is used to terminate the IAT and ILT.
+    fn create_null_thunk(&self) -> Result<NewArchiveMember> {
+        let mut buffer = Vec::new();
+
+        const NUMBER_OF_SECTIONS: usize = 2;
+        const NUMBER_OF_SYMBOLS: usize = 1;
```

Comment:
Why are these usize instead of u32 like in the original?

Reply:
I feel like it made the rest of the code cleaner (less conversions) - I can switch them back if you'd prefer.